server/odc-service/src/main/java/com/oceanbase/odc/service/session/ConnectConsoleService.java

```diff
@@ -341,6 +435,34 @@ public List<SqlExecuteResult> getAsyncResult(@NotNull String sessionId, String r
         }
     }
 
+    public AsyncExecuteResultResp getAsyncResultV2(@NotNull String sessionId, String requestId) {
+        PreConditions.validArgumentState(Objects.nonNull(requestId), ErrorCodes.SqlRegulationRuleBlocked, null, null);
+        ConnectionSession connectionSession = sessionService.nullSafeGet(sessionId);
+        AsyncExecuteContext context =
+                (AsyncExecuteContext) ConnectionSessionUtil.getExecuteContext(connectionSession, requestId);
+        boolean shouldRemoveContext = context.isFinished();
+        try {
+            List<JdbcGeneralResult> resultList = context.getFinishedSqlExecutionResults();
```

Comment:
every getResult calling get full result?

Reply:
no, only incremental results would be returned. 
Renamed this method to getMoreSqlExecutionResults

Reply:
it is robust.

thinks about this scenario, steps like:
1. front-end query getMoreResults API
2. back-end process sucess then return response to front-end
3. front-end failed to get result due timeout

then front-end may retry getMoreResults, however back-end will never return the missed result any more.

so here a cursor parameter may involved for getMoreResults API
- reponse dto may contains cursor value
- front-end set cursor while call getMoreResults API
- front-end will retry for timeout exception

Reply:
I understand what you mean. This issue is similar to TCP timeout retransmission.
1. The front-end received the result set of the first `N` SQL statements for its first request. Use the `finished` field to determine whether the execution is complete. If it is not, continue with the request;
2. The second request carries `ack=N+1`, and the backend receives it and returns `K` incremental results;
3. If the second request timeout does not return, send the third request, and the ack is still `N+1.` After receiving it, the backend returns `K` incremental results
4. If the result of the third request is received and still not completed, send the fourth request with `ack=N+K+1`

Reply:
After evaluation, it was found that the cost of doing so is relatively high. To support re-generate of the result set, it is necessary to modify the code for generating `SqlExecuteResult`. It also involves switching security context, which incurs high development costs. 

Can we proceed with the current method first, as there has been no decrease in stability compared to before.